src/history/HistoryArchive.cpp

```diff
@@ -67,6 +67,19 @@ HistoryArchiveState::futuresAllResolved() const
     return true;
 }
 
+bool
+HistoryArchiveState::futuresAllClear() const
+{
+    for (auto const& level : currentBuckets)
```

Comment:
Could be replaced by std::all_of

Reply:
(Rafal isn't reviewing this anymore, but this is fixed.)